server/util.go

```diff
@@ -206,8 +206,7 @@ func dumpBinaryTime(dur time.Duration) (data []byte) {
 
 func dumpBinaryDateTime(data []byte, t types.Time, loc *time.Location) ([]byte, error) {
 	if t.Type() == mysql.TypeTimestamp && loc != nil {
```

Comment:
before this pr, due to `loc != nil`, the `t1, err := t.GoTime(time.Local)` is useless.